src/osd/osd_types.cc

```diff
@@ -2873,7 +2873,11 @@ void pg_info_t::dump(Formatter *f) const
   f->dump_int("last_user_version", last_user_version);
   f->dump_stream("last_backfill") << last_backfill;
   f->dump_int("last_backfill_bitwise", (int)last_backfill_bitwise);
-  f->dump_stream("purged_snaps") << purged_snaps;
+  f->open_array_section("purged_snaps");
+  for (const auto&& i: purged_snaps) {
+    f->dump_stream("snap") << i;
+  }
+  f->close_section();
```

Comment:
purged_snaps is an interval set, which means it can compactly include a million snaps (1~1000000), and we don't want to dump them all.  this probably needs to do an array of objects with "start" and "length" values.